### ISSUE TITLE
Incomplete tests: mesh extension - adding VMs to a cluster

### ISSUE THREAD

**Issue by kyessenov** [istio/istio#1078]
Context: [production readiness proposal and plan](https://goo.gl/pmjuYg)

The feature “_mesh extension - adding VMs to a cluster_” is identified as incomplete test coverage. As a measure to reach Istio production quality in 0.3, this issue is opened to track the progress of providing sufficient test coverage for the feature in question.

This issue is initially assigned to the working group leads.

**Recommendations**:
- Come up with a test plan for how the feature should be tested. Identify and document (in this issue, or a separate document this issue will link to) test cases and whether each will be covered by an integration test (with some level of mocking and/or starting up selected components in test environment), or end to end test (aka system test, which brings up the whole Istio to exercise the feature; currently live in the istio/istio repo).
- If you expect the test plan is complex and controversial, consider having it formally reviewed.
- State an ETA in the issue, and assign this to a developer who will execute the test plan.
- Execute the test plan as the highest priority for 0.3.
- For any bug or defect identified by the new test cases, file a new issue and link it from this one.
- Close this issue only when the test plan is completely executed, and all additional issues identified in step 3 are fixed.

If you see a risk of not fixing this issue by end of 0.3, please contact istio-production-quality@googlegroups.com as soon as possible. We will also periodically ping the issue for update.

**Comment by ldemailly** [comment]
it'd be great to finally get this in for 0.6 at least

**Comment by jasmine-jaksic** [comment]
@hklai @costinm what is still pending on this?

**Comment by hklai** [comment]
Removing "area/test and release" label as it is not meant for adding functional tests or test coverage.
Please assign issues to the appropriate functional area going forward.

**Comment by hklai** [comment]
We added some VM test which that failed and was disabled. 
https://github.com/istio/istio/issues/4794
